river/tree/_utils/gradient_hessian.py

```diff
@@ -0,0 +1,130 @@
+import copy
+
+from river import stats
+
+
+class GradHess:
+    """ The most basic inner structure of the Streaming Gradient Trees. """
+
+    __slots__ = ["gradient", "hessian"]
+
+    def __init__(self, gradient: float = 0.0, hessian: float = 0.0, *, grad_hess=None):
+        if grad_hess:
+            self.gradient = grad_hess.gradient
+            self.hessian = grad_hess.hessian
+        else:
+            self.gradient = gradient
+            self.hessian = hessian
+
+    def __iadd__(self, other):
+        self.gradient += other.gradient
+        self.hessian += other.hessian
+
+        return self
+
+    def __isub__(self, other):
+        self.gradient -= other.gradient
+        self.hessian -= other.hessian
+
+        return self
+
+    def __add__(self, other):
+        new = copy.deepcopy(self)
+        new += other
+        return new
+
+    def __sub__(self, other):
+        new = copy.deepcopy(self)
+        new -= other
+        return new
+
+
+class GradHessStats:
+    """ Class used to monitor and update the gradient/hessian information in Streaming Gradient
+    Trees.
+
+    Represents the aggregated gradient/hessian data in a node (global node statistics), category,
+    or numerical feature's discretized bin.
+    """
+
+    def __init__(self):
+        self.x_m = stats.Mean()
+        self.g_var = stats.Var()
+        self.h_var = stats.Var()
+        self.gh_cov = stats.Cov()
+
+    def get_x(self) -> float:
+        """ Get the centroid x data that represents all the observations inside a bin. """
+        return self.x_m.get()
+
+    def __iadd__(self, other):
+        self.x_m += other.x_m
+        self.g_var += other.g_var
+        self.h_var += other.h_var
+        self.gh_cov += other.gh_cov
+
+        return self
+
+    def __isub__(self, other):
+        self.x_m -= other.x_m
+        self.g_var -= other.g_var
+        self.h_var -= other.h_var
+        self.gh_cov -= other.gh_cov
+
+        return self
+
+    def __add__(self, other):
+        new = copy.deepcopy(self)
+        new += other
+
+        return new
+
+    def __sub__(self, other):
+        new = copy.deepcopy(self)
+        new -= other
+
+        return new
+
+    def update(self, gh: GradHess, x=None, w: float = 1.0):
+        # Update x values in the case of numerical features (binning strategy)
+        if x is not None:
+            self.x_m.update(x, w)
+
+        self.g_var.update(gh.gradient, w)
+        self.h_var.update(gh.hessian, w)
+        self.gh_cov.update(gh.gradient, gh.hessian, w)
+
+    def mean(self) -> GradHess:
```

Comment:
`mean`, `variance`, and `covariance` could be properties.

Reply:
Good catch! I made them properties in commit [940122d](https://github.com/online-ml/river/pull/458/commits/940122dfb81b6e62d7f43a2e51cfb113b90d899d).